_posts/2021-02-20-kotlin-use-site-targets.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title:  "Unwrapping Use-site Targets in Kotlin"
+date:   2021-03-02 10:00:00 +0200
+comments: true
+img:
+  href: 2021-03-02-dogs-of-berlin.png
+  copyright: Dogs of Berlin by Christian Alvart
+  alt: Dogs of Berlin
+---
+I like Kotlin for its expressiveness. Its interoperability with Java
+allows to easily use the existing Java ecosystem while enjoying a modern
+language. When it comes to annotations Kotlin handles them almost the same
+as Java. In Kotlin, a single syntax construction might be compiled into
+several JVM constructions. A Kotlin property would produce a JVM field,
+a getter, and optionally a setter. This is where annotation use-site targets
+are useful.
+
+{% include picture.html %}
+
+Take a look at the following code:
+
+```kotlin
+data class User(
+    @Unique
+    val name: String,
+    @Size(min = 18, max = 100)
+    val age: Int,
+)
+```
+
+Once it is compiled, the corresponding class would have a constructor,
+fields, getters, equals, hashcode, and some other language elements.
+If you need that in the compiled code annotations were generated specifically
+for getters or fields you can specify so-called use-site targets:
```

Comment:
...fields. You can specify so-called use-site targets:

Reply:
@rebeccaandersonediting I guess it changes the meaning of the sentence. The "you can specify..." part is related to the previous statement  staring with "If you need". 

It is like "If you need ... **then** you can specify...". Could you please rephrase it to make clear?

Reply:
"If you need that in compiled code, you can specify so-called use-site targets. Annotations were generated specifically for getters or fields to allow for this."  OR "If you need that in compiled code, annotations generated specifically for getters or fields allow you to specify so-called use-site targets."

Reply:
@rebeccaandersonediting these options also change the meaning. I am not sure if I am able to explain it properly without much context, but let me try. If a user needs "to generate annotations specifically for getters or fields" then the user can "specify so-called use-site targets". In other words, specifying use-site targets will allow the user to have annotations generated specifically for getters or setters in the compiled code.

Does it make sense?

Reply:
If you need that in compiled code, you must generate annotations specifically for getters or fields, which will then allow you to specify so-called use-site targets.

Reply:
"If you need that in the compiled code, annotations must be generated specifically for getters or fields in order to allow you to specify so-called use-site targets"

Reply:
By generating annotations specifically for getters or fields, you will be able to include that in the compiled code and specify so-called use-site targets.

Reply:
"If you need that in the compiled code, you must specify the so-called use-site targets in order to generate annotations specifically for getters or fields."